docs/modules/tests.md

```diff
@@ -0,0 +1,15 @@
+# Module: Tests
+
+All modules **MUST** have tests with minimum coverage.
```

Comment:
I question `MUST`.  Some tests for modules take close to an hour to run.  I'd worry that a full test suite might really slow stuff down too much.

I'm not sure how to weight the pros and cons of this.

Reply:
Yeah this is a tricky one. While I think some test are important, I think we should probably define what a good test looks like.

Reply:
updating it the following statement:

`All modules **SHOULD** try to have tests with minimum coverage.
`

Reply:
@brianpham and @greenkiwi I have revisited few things and added some tips on how to build a tests. So let me know how it looks now